conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinIndexer.scala

```diff
@@ -0,0 +1,86 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import cats.effect.{ContextShift, ExitCode, IO}
+import com.typesafe.scalalogging.LazyLogging
+import org.http4s.headers.Authorization
+import org.http4s.BasicCredentials
+import org.http4s.client.blaze.BlazeClientBuilder
+
+import tech.cryptonomic.conseil.indexer.config.{
+  BatchFetchConfiguration,
+  HttpStreamingConfiguration,
+  LorreConfiguration,
+  NetworkCallsConfiguration
+}
+import tech.cryptonomic.conseil.common.config.Platforms
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinConfiguration
+import tech.cryptonomic.conseil.indexer.LorreIndexer
+import tech.cryptonomic.conseil.indexer.logging.LorreProgressLogging
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+import tech.cryptonomic.conseil.indexer.bitcoin.streams.BlocksStream
+
+/** * Class responsible for indexing data for Tezos BlockChain */
+class BitcoinIndexer(
+    lorreConf: LorreConfiguration,
+    bitcoinConf: BitcoinConfiguration,
+    callsConf: NetworkCallsConfiguration,
+    streamingClientConf: HttpStreamingConfiguration,
+    batchingConf: BatchFetchConfiguration
+) extends LazyLogging
+    with LorreIndexer
+    with LorreProgressLogging {
+
+  // Use global execution context only for the PoC
+  implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
+  private val httpEC = ExecutionContext.global
+
+  /**
+    * Produces the `IO` to be run as an indexer.
+    *
+    * @return the [[cats.effect.ExitCode]] the JVM exits with
+    */
+  def run: IO[ExitCode] = {
```

Comment:
I wouldn't expose the exit-code at this internal component level.
To me it looks like a top-level artefact only, that shouldn't be coupled at this layer.

Reply:
Right point, I would change it if I had the opportunity :)